docs/OffsetHelper.md

```diff
@@ -333,6 +333,562 @@ Retire the specified TCO2 tokens.
+  using MATIC, specifying the exact amount of TCO2s to retire,
+- `autoOffsetExactInETH()` if the user would like to perform a retirement
+  using MATIC, swapping all sent MATIC into TCO2s,
+- `autoOffsetExactOutToken()` if the user would like to perform a retirement
+  using an ERC20 token (USDC, WETH or WMATIC), specifying the exact amount
+  of TCO2s to retire,
+- `autoOffsetExactInToken()` if the user would like to perform a retirement
+  using an ERC20 token (USDC, WETH or WMATIC), specifying the exact amount
+  of token to swap into TCO2s.
+
+In these methods, "auto" refers to the fact that these methods use
+`autoRedeem()` in order to automatically choose a TCO2 token corresponding
+to the oldest tokenized carbon project in the specfified token pool.
+There are no fees incurred by the user when using `autoRedeem()`, i.e., the
+user receives 1 TCO2 token for each pool token (BCT/NCT) redeemed.
+
+There are two `view` helper functions `calculateNeededETHAmount()` and
+`calculateNeededTokenAmount()` that should be called before using
+`autoOffsetExactOutETH()` and `autoOffsetExactOutToken()`, to determine how
+much MATIC, respectively how much of the ERC20 token must be sent to the
+`OffsetHelper` contract in order to retire the specified amount of carbon.
+
+The two `view` helper functions `calculateExpectedPoolTokenForETH()` and
+`calculateExpectedPoolTokenForToken()` can be used to calculate the
+expected amount of TCO2s that will be offset using functions
+`autoOffsetExactInETH()` and `autoOffsetExactInToken()`.
+
+### constructor
+
+```solidity
+constructor(string[] _eligibleTokenSymbols, address[] _eligibleTokenAddresses) public
+```
+
+Contract constructor. Should specify arrays of ERC20 symbols and
+addresses that can used by the contract.
+
+_See `isEligible()` for a list of tokens that can be used in the
+contract. These can be modified after deployment by the contract owner
+using `setEligibleTokenAddress()` and `deleteEligibleTokenAddress()`._
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| _eligibleTokenSymbols | string[] | A list of token symbols. |
+| _eligibleTokenAddresses | address[] | A list of token addresses corresponding to the provided token symbols. |
+
+### Redeemed
+
+```solidity
+event Redeemed(address who, address poolToken, address[] tco2s, uint256[] amounts)
+```
+
+Emitted upon successful redemption of TCO2 tokens from a Toucan
+pool token such as BCT or NCT.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| who | address | The sender of the transaction |
+| poolToken | address | The address of the Toucan pool token used in the redemption, for example, NCT or BCT |
+| tco2s | address[] | An array of the TCO2 addresses that were redeemed |
+| amounts | uint256[] | An array of the amounts of each TCO2 that were redeemed |
+
+### onlyRedeemable
+
+```solidity
+modifier onlyRedeemable(address _token)
+```
+
+### onlySwappable
```

Comment:
I noticed that "swappable" is spelt correctly here, but not with `isSwapable` . Would be nice to fix that, although probably need to keep the mispelt function for backwards compatibility.